python/ray/tune/trial_runner.py

```diff
@@ -149,6 +182,7 @@ def __init__(self,
         self._remote_checkpoint_dir = remote_checkpoint_dir
         self._syncer = get_cloud_syncer(local_checkpoint_dir,
                                         remote_checkpoint_dir, sync_to_cloud)
+        self._stopper = stopper if issubclass(stopper, StopperClass) else None
```

Comment:
for backwards compatibility maybe we should continue to support non-subclasses and just log a soft deprecation warning to extend `StopperClass`

Reply:
Yes; this PR doesn't break backwards-compat. This is in the trial_runner (and is specifically to stop _all_trials). See `trial.py` for the standard trial-level stopping.

Reply:
oh right another possibly cleaner thing to do is in `Experiment` wrap any function-based stoppers with `FuncStopper` or something, calling the function in __call__ (and with stop_all returning False by default), so you dont need to handle this case specifically in trial_runner. ie you can remove the check here and in _stop_experiment_if_needed

i dont feel strongly either way though so as its written is also fine